### PR TITLE
Flutter for iOS — Page contents (squashed)

### PR DESCRIPTION
**Pull request by rock3r**
This PR contains all the work from #965 #977 #978 #979 #980 #981 #982 #984 #985 #986 #987 — we want one big PR to review @niamhpower's work. She's still the main author of the contents, and has signed the CLA, but something is wrong and won't really work with the bot, so I am acting as a proxy here.

I have addressed some of the comments on #979 and me and Niamh will bring on with the rest of the comments which aren't applicable to the Android page too (which is the blueprint for the iOS one). We'll bring on the further changes to both the Android and iOS pages in future PRs as agreed with @mjohnsullivan.

### CODE REVIEW

File: flutter-for-ios.md [hunk 1]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+differences to a  `UIView`. To start, widgets only exist for one frame, and on
+every new frame, Flutter’s framework creates a new tree of widget instances.
+In comparison, an iOS view is drawn once and does not redraw until it is
+invalidated using `setNeedsDisplay()`.
+
+Furthermore, unlike `UIView`, Flutter’s widgets are immutable, which allows
+them to be very lightweight. By default, Flutter’s widgets are styled as
+‘Material’ (i.e. Android themed), however you can use [Cupertino
+widgets](https://flutter.io/widgets/cupertino/) to specify an iOS theme if you
+require.
```
Comment (Sfshaza) [inline flutter-for-ios.md]: if you require -> if required.

File: flutter-for-ios.md [hunk 2]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+
+In Flutter, the equivalent to this is a `Widget`. However, these have a few
+differences to a  `UIView`. To start, widgets only exist for one frame, and on
+every new frame, Flutter’s framework creates a new tree of widget instances.
+In comparison, an iOS view is drawn once and does not redraw until it is
+invalidated using `setNeedsDisplay()`.
+
+Furthermore, unlike `UIView`, Flutter’s widgets are immutable, which allows
+them to be very lightweight. By default, Flutter’s widgets are styled as
+‘Material’ (i.e. Android themed), however you can use [Cupertino
```
Comment (Sfshaza) [inline flutter-for-ios.md]: as `Material` components (Android themed), ...

File: flutter-for-ios.md [hunk 3]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+`StatelessWidgets` are useful when the part of the user interface you are
+describing does not depend on anything other than the initial configuration
+information in the widget.
+
+For example, in iOS, this would be similar to placing a `UIImageView` with
+your logo as the `image`. If the logo is not going to change during runtime,
+you would use a `StatelessWidget` in Flutter.
+
+If you want to dynamically change the UI based on data received after making an
+HTTP call then you would use a `StatefulWidget`. After the HTTP call had
```
Comment (Sfshaza) [inline flutter-for-ios.md]: had complete => completed

File: flutter-for-android.md [hunk 4]
```diff
@@ -1018,7 +1018,7 @@ You can use it by adding it to your dependencies in `pubspec.yaml`:
 {% prettify yaml %}
 dependencies:
   ...
-  http: '>=0.11.3+16'
+  http: "^0.11.3+16"
```
Comment (Sfshaza) [inline flutter-for-android.md]: We prefer no quotes or single quotes in pubspec.

File: flutter-for-ios.md [hunk 5]
```diff
@@ -4,12 +4,6 @@ title: Flutter for iOS Developers
 permalink: /flutter-for-ios/
 ---
 
-# This document is a work in progress
-We're working on this document right now. Please do not rely on its completeness
-and accuracy until it's finished and it's shown in the sidebar as a link.
-
----
-
 This document is meant for iOS developers looking to apply their
```
Comment (Sfshaza) [inline flutter-for-ios.md]: delete "meant"

File: flutter-for-ios.md [hunk 6]
```diff
@@ -4,12 +4,6 @@ title: Flutter for iOS Developers
...
 existing iOS knowledge to build mobile apps with Flutter. If you understand
 the fundamentals of the iOS framework then you can use this document as a
```
Comment (Sfshaza) [inline flutter-for-ios.md]: you can use this document as a ???

File: flutter-for-ios.md [hunk 7]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new Center(child: new Text(textToShow)),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: _updateText,
+        tooltip: 'Update Text',
+        child: new Icon(Icons.update),
```
Comment (Sfshaza) [inline flutter-for-ios.md]: This code is shown as italic, not codefont. That could be a GitHub hiccup, as there are variable names that begin with underscore. There is no staged version provided, or I would check this.

File: flutter-for-ios.md [hunk 8]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+## How do I layout my widgets? Where is my Storyboard? 
+
+In iOS, you may use a Storyboard file to organise your views and set
```
Comment (Sfshaza) [inline flutter-for-ios.md]: According to the Google Style  Guide, our docs use American spelling, not British:

organise => organize

File: flutter-for-ios.md [hunk 9]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+}
+{% endprettify %}
+
+Here, `dataLoader()` is the `Isolate` that runs in its own separate execution thread.
+In this isolate you can do more CPU intensive processing, for example parsing a big
+JSON, or doing computationally intensive math such as encryption or signal processing.
+
+A full example that you can run is below.
+
+<!-- skip -->
```
Comment (Sfshaza) [inline flutter-for-ios.md]: If it's a full example, don't use skip. We want the analyzer to check it. We only use that for partial code that will fail in the analyzer.

File: flutter-for-ios.md [hunk 10]
```diff
@@ -27,3 +21,1847 @@ that are most relevant to your needs.
...
+[SQFlite](https://pub.dartlang.org/packages/sqflite).
+
+# Notifications
+
+## How do I setup Push Notifications?
+
+In iOS, you would need to register your app on the developer portal to allow
+push notifications. In Flutter, you can access this functionality by using the
+`Firebase_Messaging` plugin. You can find full instructions on how to get set
+up [here](https://pub.dartlang.org/packages/firebase_messaging).
```
Comment (Sfshaza) [inline flutter-for-ios.md]: Never link on "here". There are many reasons, but foremost it is not accessible-compliant. 
https://developers.google.com/style/link-text   and 
https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/

Better: See the [firebase_messaging](https://pub.dartlang.org/packages/firebase_messaging) plugin for more information on using the Firebase Cloud Messaging API.

File: flutter-for-ios.md (hunk 9 above)
Comment (rock3r) [reply flutter-for-ios.md]: We "took inspiration" (as in, lifted this) from the "Flutter for Android developers" page. Should we have it removed on Android too?

File: flutter-for-ios.md (hunk 8 above)
Comment (rock3r) [reply flutter-for-ios.md]: Sorry, must've slipped through (Brits here!)

File: flutter-for-ios.md (hunk 7 above)
Comment (rock3r) [reply flutter-for-ios.md]: Looks fine running Jekyll locally:

image

**Comment by rock3r** [comment]
Most comments should be addressed, here and in the other squashed PRs. This should be ok for at least an initial merge, and we can pick it up again later if anything else is needed.

**Comment by Sfshaza** [comment]
Much better! Any other minor copyedits can be handled later. LGTM

**Pull request merged**